src/services/refactors/inlineVariable.ts

```diff
@@ -0,0 +1,236 @@
+
+const refactorName = "Inline variable";
+const refactorDescription = getLocaleSpecificMessage(Diagnostics.Inline_variable);
+
+const inlineVariableAction = {
+    name: refactorName,
+    description: refactorDescription,
+    kind: "refactor.inline.variable"
+};
+
+interface InliningInfo {
+    references: Node[];
+    declaration: VariableDeclaration;
+    replacement: Expression;
+}
+
+registerRefactor(refactorName, {
+    kinds: [inlineVariableAction.kind],
+
+    getAvailableActions(context) {
+        const {
+            file,
+            program,
+            preferences,
+            startPosition,
+            triggerReason
+        } = context;
+
+        // tryWithReferenceToken is true below when triggerReason === "invoked", since we want to
+        // always provide the refactor in the declaration site but only show it in references when
+        // the refactor is explicitly invoked.
+        const info = getInliningInfo(file, startPosition, triggerReason === "invoked", program);
+        if (!info) {
+            return emptyArray;
+        }
+
+        if (!refactor.isRefactorErrorInfo(info)) {
+            return [{
+                name: refactorName,
+                description: refactorDescription,
+                actions: [inlineVariableAction]
+            }];
+        }
+
+        if (preferences.provideRefactorNotApplicableReason) {
+            return [{
+                name: refactorName,
+                description: refactorDescription,
+                actions: [{
+                    ...inlineVariableAction,
+                    notApplicableReason: info.error
+                }]
+            }];
+        }
+
+        return emptyArray;
+    },
+
+    getEditsForAction(context, actionName) {
+        Debug.assert(actionName === refactorName, "Unexpected refactor invoked");
+
+        const { file, program, startPosition } = context;
+
+        // tryWithReferenceToken is true below since here we're already performing the refactor.
+        // The trigger kind was only relevant when checking the availability of the refactor.
+        const info = getInliningInfo(file, startPosition, /*tryWithReferenceToken*/ true, program);
+        if (!info || refactor.isRefactorErrorInfo(info)) {
+            return undefined;
+        }
+
+        const { references, declaration, replacement } = info;
+        const edits = textChanges.ChangeTracker.with(context, tracker => {
+            for (const node of references) {
+                tracker.replaceNode(file, node, getReplacementExpression(node, replacement));
+            }
+            tracker.delete(file, declaration);
+        });
+
+        return { edits };
+    }
+});
+
+function getInliningInfo(file: SourceFile, startPosition: number, tryWithReferenceToken: boolean, program: Program): InliningInfo | RefactorErrorInfo | undefined {
+    const checker = program.getTypeChecker();
+    const token = getTokenAtPosition(file, startPosition);

```

Comment:
The refactor still doesn't allow users to trigger from the right edge

```ts
const x/**/ = 10;
const y = x;
```

You need to switch this to something else like `getTouchingPropertyName`.